pkg/spoke/controllers/deletioncontroller/deletion_controller.go

```diff
@@ -117,36 +134,68 @@ func (m *StaleManifestDeletionController) syncManifestWork(ctx context.Context,
 		return nil
 	}
 
-	// delete applied resources which are no longer maintained by manifest work
-	noLongerMaintainedResources := findUntrackedResources(manifestWork.Status.AppliedResources, appliedResources)
-	if errs := m.deleteAppliedResources(noLongerMaintainedResources); len(errs) != 0 {
-		return utilerrors.NewAggregate(errs)
-	}
-
-	// update work status with latest applied resources. if this conflicts, we'll simply try again later
+	// update work status with latest applied resources. if this conflicts, we'll try again later
 	// for retrying update without reassessing the status can cause overwriting of valid information.
 	manifestWork.Status.AppliedResources = appliedResources
 	_, err := m.manifestWorkClient.UpdateStatus(ctx, manifestWork, metav1.UpdateOptions{})
-	return err
+	if err != nil {
+		return err
+	}
+
+	// requeue the work if there exists deleting applied resources
+	if len(deletingResources) != 0 {
```

Comment:
Doing it this way prevents orphaning of resources and prevents adoption by other manifest work without an outage.  This can result in perma-stuck resources.  Be sure you mean this.

Reply:
To fix this issue, I suggest adding a new optional property, like `DeletionTimestamp`, into AppliedManifestResourceMeta struct. The property will be set once controller calls the delete api for the applied resource successfully. StaleManifestDeletionController will only delete applied resource of stale manifest whose `DeletionTimestamp` is not set, which ensures deletion of applied resource is done only once by controller.